pkg/endpoint/regeneration/wait.go

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package regeneration
+
+import (
+	"context"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// InitWaitGroup for delaying the initial endpoint regeneration.
+type InitWaitGroup interface {
+	// Add a wait function to delay the initial endpoint regeneration.
+	Add(name string, waitFn lock.InitWaitFunc)
+
+	// Wait for all registered initializers to complete.
+	Wait(context.Context) error
+}
+
+func NewInitWaitGroup(lc cell.Lifecycle) InitWaitGroup {
```

Comment:
The naming here is definitely confusing. Maybe we come up with some kind of naming convention like BeforeRegeneration for each synchronization point?

Reply:
Agreed that the naming isn’t great. Will try to come up with something better today.